arch/arm64/src/common/arm64_boot.c

```diff
@@ -100,26 +106,28 @@ void arm64_boot_el2_init(void)
 {
   uint64_t reg;
 
-  reg   = read_sysreg(sctlr_el2);
-  reg   |= (SCTLR_EL2_RES1 |    /* RES1 */
-            SCTLR_I_BIT |       /* Enable i-cache */
-            SCTLR_SA_BIT);      /* Enable SP alignment check */
+  reg = read_sysreg(sctlr_el2);
+  reg |= (SCTLR_EL2_RES1 |    /* RES1 */
+#ifndef CONFIG_ARM64_ICACHE_DISABLE
```

Comment:
In general I do not support such usage of Kconfig options. It is misleading, by stating "Disable XXX at __start", but actual usage is "do not enable and rely on defaults before start". So if there is the underlying BL that enabled I-Cache even if the user selects "Disable I-Cache at __start" and actually expects that I-Cache will be disabled will get the I-Cache enabled at the end.

Reply:
@pkarashchenko I have deleted the third commit and will resubmit it after I come up with a more appropriate modification. Please help review the current two commits.